### PR TITLE
Add support for renaming own collections

### PR DESCRIPTION
**Pull request by ua741**
## Description
Ability to change name of an album

## Test Plan
- [x] Feature working as expected. 
- [x] Changed the name of screenshot collection and verified that the new screenshots are being uploaded on the same collection.

### CODE REVIEW

File: lib/ui/gallery_app_bar_widget.dart [hunk 1]
```diff
@@ -107,6 +115,37 @@ class _GalleryAppBarWidgetState extends State<GalleryAppBarWidget> {
...
+      return;
+    }
+    var result = await showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return ChangeCollectionNameDialog(name: appBarTitle);
+      },
+      barrierColor: Colors.black.withOpacity(0.85),
+    );
+    if (result == null) {
```
Comment (vishnukvmd) [inline lib/ui/gallery_app_bar_widget.dart:129]: Why would result be `null`?

File: lib/ui/gallery_app_bar_widget.dart [hunk 2]
```diff
@@ -107,6 +115,37 @@ class _GalleryAppBarWidgetState extends State<GalleryAppBarWidget> {
     );
   }
 
+  Future<dynamic> _renameAlbum(BuildContext context) async {
+    if (widget.type != GalleryAppBarType.owned_collection) {
+      return;
+    }
+    var result = await showDialog(
```
Comment (vishnukvmd) [inline lib/ui/gallery_app_bar_widget.dart:122]: `final` by default

File: lib/ui/gallery_app_bar_widget.dart [hunk 3]
```diff
@@ -54,6 +57,7 @@ class _GalleryAppBarWidgetState extends State<GalleryAppBarWidget> {
   final _logger = Logger("GalleryAppBar");
   StreamSubscription _userAuthEventSubscription;
   Function() _selectedFilesListener;
+  String appBarTitle;
```
Comment (vishnukvmd) [inline lib/ui/gallery_app_bar_widget.dart:60]: `_appBarTitle` since it's a private variable.

File: lib/ui/change_collection_name_dialog.dart [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  @override
+  void initState() {
+    super.initState();
+    _newCollectionName = widget.name;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      title: Text("enter new album name"),
```
Comment (vishnukvmd) [inline lib/ui/change_collection_name_dialog.dart:31]: `enter a new name`

File: lib/ui/change_collection_name_dialog.dart [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            ),
+          ),
+          onPressed: () {
+            if (_newCollectionName.trim().isEmpty) {
+              showErrorDialog(
+                  context, "empty name", "album name cannot be empty");
+              return;
+            }
+            if (_newCollectionName.trim().length > 100) {
+              showErrorDialog(context, "too large",
```
Comment (vishnukvmd) [inline lib/ui/change_collection_name_dialog.dart:84]: `name too large`

File: lib/ui/gallery_app_bar_widget.dart [hunk 6]
```diff
@@ -107,6 +115,37 @@ class _GalleryAppBarWidgetState extends State<GalleryAppBarWidget> {
...
+      builder: (BuildContext context) {
+        return ChangeCollectionNameDialog(name: appBarTitle);
+      },
+      barrierColor: Colors.black.withOpacity(0.85),
+    );
+    if (result == null) {
+      return;
+    }
+
+    final dialog = createProgressDialog(context, "changing name...");
```
Comment (vishnukvmd) [inline lib/ui/gallery_app_bar_widget.dart:133]: Could perhaps move this logic into `ChangeCollectionNameDialog` to reduce the logic on the app bar? Not sure.

File: lib/ui/gallery_app_bar_widget.dart (hunk 1 above)
Comment (ua741) [reply lib/ui/gallery_app_bar_widget.dart:129]: When user cancels the dialog or don't change the name, we are returning null to indicate no action is required.

File: lib/ui/gallery_app_bar_widget.dart (hunk 6 above)
Comment (ua741) [reply lib/ui/gallery_app_bar_widget.dart:133]: In future, during UX revamp, I was thinking of making this `ChangeCollectionNameDialog` a generic user input dialog, which can use reused. That's why I kept it separate for now.

File: lib/ui/gallery_app_bar_widget.dart (hunk 1 above)
Comment (vishnukvmd) [reply lib/ui/gallery_app_bar_widget.dart:129]: I thought I had deleted this comment.

**Review by vishnukvmd — APPROVED** [review]

**Pull request merged**